File: 2024-mk2/linked-list/linked-list-cycle.go
* Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeNthFromEnd(head *ListNode, n int) *ListNode {
    // Brute force approach
    tmpHead := head
    l := []*ListNode{}
    // Create an array of ListNodes
    for head != nil {
        l = append(l, head)
        head = head.Next
    }
    // Calculate item to remove
    remove := len(l)-n
    // If the element is the first in the array, remove the Next (will be nil if it is alone)
    if remove == 0 {
        return tmpHead.Next
    }
    // Find the element BEFORE the element to remove, and set it's next to the item to remove's next
    l[remove-1].Next = l[remove].Next

    
    return tmpHead
}

File: 2024-mk2/linked-list/linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        m = {}
        cur = head

        while cur != None:
            if cur in m:
                return True
            m[cur] = True
            cur = cur.next

        return False
        
        #fast, slow = head, head
#
        #while fast != None and fast.next != None:
        #    fast = fast.next.next
        #    slow = slow.next
        #    if fast == slow:
        #        return True
#
        #return False

